docs/contributing/coding-guidelines/ros-nodes/class-design.md

```diff
@@ -1,5 +1,68 @@
 # Class design
 
+We'll use the `autoware_gnss_poser` package as an example.
+
+## Namespaces
+
+```cpp
+namespace autoware::gnss_poser
+{
+...
+} // namespace autoware::gnss_poser
+```
+
+- Everything should be under `autoware::gnss_poser` namespace.
+- Closing braces should contain a comment with the namespace name.
+
+## Classes
+
+### Nodes
+
+#### `gnss_poser_node.hpp`
+
+```cpp
+class GNSSPoserNode : public rclcpp::Node
+{
+  public:
+    explicit GNSSPoserNode(const rclcpp::NodeOptions & node_options);
+  ...
+}
+```
+
+#### `gnss_poser_node.cpp`
+
+```cpp
+GNSSPoserNode::GNSSPoserNode(const rclcpp::NodeOptions & node_options)
+: Node("gnss_poser", node_options)
+{
+  ...
+}
+```
+
+- The class name should be in `CamelCase`.
+- Node classes should inherit from `rclcpp::Node`.
+- The constructor should be explicit.
```

Comment:
s/should/must/

Reply:
Done with https://github.com/autowarefoundation/autoware-documentation/pull/547/commits/fea457f9918457fb8c5eb22adb2ec92ee89e8928